system/metrics/sinks/elastic/elasticSink.cpp

```diff
@@ -320,7 +357,54 @@ bool ElasticMetricSink::prepareToStartCollecting()
 
 void ElasticMetricSink::doCollection()
 {
+    nlohmann::json reportData;
+    auto reportMetrics = pManager->queryMetricsForReport(name);
+    for (auto &pMetric: reportMetrics)
+    {
+        auto metricType = pMetric->queryMetricType();
 
+        if (metricType == METRICS_HISTOGRAM)
+        {
+            if (pMetric->queryValue() || !ignoreZeroMetrics)
+            {
+                auto values = pMetric->queryHistogramBucketLimits();
+                auto counts = pMetric->queryHistogramValues();
+                const std::string &histogramName = getMetricReportName(pMetric);
+
+                // retrieve the inf value and remove it from the vector so the counts and values sizes match
+                auto inf = counts.back();
+                counts.pop_back();
+
+                reportData[histogramName]["values"] = values;
+                reportData[histogramName]["counts"] = counts;
+
+                // add the inf value if indicated
+                if (!ignoreZeroMetrics || inf)
+                {
+                    // create an inf name that looks like <name>_<histogramSuffix>_inf<countMetricSuffix>
+                    std::string infName = histogramName;
+                    infName.append("_inf").append(countMetricSuffix);
+                    reportData[infName] = inf;
+                }
+            }
+        }
+        else
+        {
+            auto value = pMetric->queryValue();
+            if (value || !ignoreZeroMetrics)
+                reportData[getMetricReportName(pMetric)] = pMetric->queryValue();
+        }
+    }
+
+    std::string json = reportData.dump();
+
+    // Index if report data is not empty
+    if (!json.empty())
+    {
+        std::string endpoint;
+        endpoint.append("/").append(indexName.str()).append("/_doc");
+        auto resp = pClient->Post(endpoint.c_str(), elasticHeaders, json, "application/json");
```

Comment:
I'm not very familiar w/ this client, but typically http requests require error handling, if nothing else report critical errors.
Also, perhaps outside the scope of this issue, and we might have discussed elsewhere, but I want to make sure we set appropriate connect/read/write timeouts. I assume we want a short timeout leash on all metric reporting (have you seen any official guidance on this?)